File: C++Code/Sorting/quickSort.cpp
#include<iostream>
using namespace std;
int partition(int arr[],int low,int high)
{
    int pivot=arr[high-1];
    int j=low-1;
    for(int i=low;i<high;i++){
        if(pivot>arr[i]){
            j++;
            int tem=arr[j];
            arr[j]=arr[i];
            arr[i]=tem;
        }
    }
    int tem=arr[j+1];
    arr[j+1]=arr[high-1];
    arr[high-1]=tem;
    return j+1;
}
void quickSort(int arr[],int low,int high){
       if (low < high)
    {  
        int pi = partition(arr, low, high);
        quickSort(arr, low, pi - 1);
        quickSort(arr, pi + 1, high);
    }
}
int main(){
    int arr[]={9,1,55,3,41,44};
    int n=sizeof(arr)/sizeof(arr[0]);
    quickSort(arr,0,n);
    for(int i=0;i<n;i++){
        cout<<arr[i]<<endl;
    }
}

File: JavaScritpt/Arrays/Sorting/QuickSort.js
function Swap(arr,i,j){
    var tem=arr[i];
    arr[i]=arr[j];
    arr[j]=tem;
}
function Partition(arr,low,high){
    var pivot=arr[high];
    var i=low-1;
    for(let j=low;j<=high-1;j++){
        if(arr[j]<pivot){
            i++;
            Swap(arr,i,j);
        }
    }
    Swap(arr,i+1,high)
    return (i+1);
}
function QuickSort(arr,low,high){
    if(low<high){
    var pi=Partition(arr,low,high);
    QuickSort(arr,low,pi-1);
    QuickSort(arr,pi+1,high);
}
}
function printArray(arr,n){
    for(let i=0;i<n;i++){
        console.log(arr[i]);
    }
}
var arr = [ 10, 7, 8, 9, 1, 5 ];
    var n = arr.length;
     
    QuickSort(arr, 0, n-1);
   
    printArray(arr, n);

function Pivot(arr,low,high){
    let pivot=arr[high]
    let i=low -1
    for(let j=low;j<high;j++){
        if(arr[j]<pivot){
            i++;
            Swap(arr,i,j)
        }
    }
    Swap(arr,i+1,high)
    return i+1;

}
    function QuickSort1(arr,low,high){
        if(low<high){
            let pi=Pivot(arr,low,high)
            QuickSort1(arr,low,pi-1)
            QuickSort1(arr,pi+1,high)
        }
    }